nixos/modules/config/no-x-libs.nix

```diff
@@ -35,7 +35,7 @@ with lib;
       networkmanager_pptp = pkgs.networkmanager_pptp.override { withGnome = false; };
       networkmanager_vpnc = pkgs.networkmanager_vpnc.override { withGnome = false; };
       networkmanager_iodine = pkgs.networkmanager_iodine.override { withGnome = false; };
-      pinentry = pkgs.pinentry.override { gtk2 = null; qt4 = null; };
+      pinentry = pkgs.pinentry.override { gtk2 = null; };
```

Comment:
Just set `pinentry = pkgs.pinentry_ncurses`.

Reply:
Won’t that be infinite recursion?

Reply:
Right, I'd define `pinentry` as an override of `pinentry_ncurses` in `all-packages.nix` to solve it.

Reply:
Right